### PR TITLE
Mission image revamp

### LINKED ISSUE #261: PTN Heaven's gate
`Command raised an exception: OperationalError: near "S": syntax error`

Occurs even after renaming channel and carrier to remove or escape `
Didn't happen pre-mission done consolidation

### LINKED ISSUE #271: New mission image format
m.carrier_image:
- Check aspect ratio and crop if not 16:9
- Resize to fit slot
- save as either shortname or channelname.png

mission image generator:
- if no carrier image exists:
   - select random from image library
   - resize to fit slot
   - save as carrier image
- use template as background
- apply carrier image as foreground with correct positioning
- overlay text as before

### PR DESCRIPTION
**Pull request by cimspin**
Fixes #261 
Closes #271 

Quite extensive changes to the way carrier mission images are handled. I think I caught everything...

### CODE REVIEW

File: MissionAlertBot.py [hunk 1]
```diff
@@ -974,6 +992,33 @@ async def gen_mission(ctx, carrier_name_search_term: str, commodity_search_term:
     if not carrier_data:
         return await ctx.send(f"No carrier found for {carrier_name_search_term}. You can use `/find` or `/owner` to search for carrier names.")
 
+    # check if the carrier has an associated image
+
+    if os.path.isfile(f"images/{carrier_data.carrier_short_name}.png"):
+        print("Carrier mission image found, checking size...")
+        image = Image.open(f"images/{carrier_data.carrier_short_name}.png")
+        image_is_good = True if image.size == (506, 285) else False
```
Comment (gcruicks) [inline MissionAlertBot.py:1000]: You could write this as:

`image_is_good = image.size == (506, 285)`

It is implicitly True or False

**Review by gcruicks — APPROVED** [review]
Code paths look ok, only comment is on some bool checks that could be re-written cleaner

File: MissionAlertBot.py (hunk 1 above)
Comment (cimspin) [reply MissionAlertBot.py]: Changed this and added a couple of small tweaks:
- default image option
- carrier_add ID format check (thanks BoozeBot!)
- carrier_add common info embed format on success

**Review by gcruicks — APPROVED** [review]

**Comment by cimspin** [comment]
- Choose default image randomly from selection
- Check if carrier exists before proceeding with image upload (whoops)
- Fixed a missing image_exists define
- Check to see if file is an image when user uploads an attachment
- Updated help description

**Review by gcruicks — APPROVED** [review]

**Pull request merged**